pkg/storage/ingest/fetcher.go

```diff
@@ -443,6 +459,9 @@ func (r *concurrentFetchers) parseFetchResponse(ctx context.Context, startOffset
 		fetchedBytes = sumRecordLengths(partition.Records)
 	}
 
+	nv := r.bufferedFetchedRecords.Add(int64(len(partition.Records)))
```

Comment:
Looks a bit weird incrementing it in the parse function. Have you considered incrementing it right after the call to `r.fetchSingle()` in `concurrentFetchers.run()`?

Reply:
I originally had it there - but changed it because it seemed semantically correct here to move it to the first place where we have actually records in memory.

I'll change it back because that makes my original hunch correct that right after `fetchSingle` it is the most logical place where one would go look.

Reply:
Done in https://github.com/grafana/mimir/pull/9855/commits/17021962c99417bf2c1dcb6afc3d60d399156456